packages/daemon/src/context/ContextMutations.js

```diff
@@ -269,6 +270,108 @@ export default class ContextMutations {
     await openVault.save()
   }
 
+  // Comms
+
+  async publishAppData(
+    appID: ID,
+    contactID: ID,
+    key: string,
+    value: Object,
+  ): Promise<void> {
+    const { openVault, io } = this._context
+    const res = openVault.identities.getContactByID(contactID) || {}
+    if (!res) {
+      throw new Error('Contact not found')
+    }
+    const { userID, contact } = res
+
+    const app = openVault.apps.getByID(appID)
+    if (!app) {
+      throw new Error('App not found')
+    }
+    const sharedAppID = app.mfid
+
+    const existingSharedData = openVault.contactAppData.getSharedData(
+      sharedAppID,
+      contactID,
+    )
+    const sharedData = existingSharedData
+      ? existingSharedData
+      : openVault.contactAppData.createSharedData(sharedAppID, contactID)
```

Comment:
If it satisfies the use case, an alternative here would be to make `getSharedData()` call `createSharedData()` or create it directly so the consumer of `getSharedData()` never needs the additional logic.

Reply:
`ContextQueries.getContactAppFeeds()` requires `getSharedData()` without creation. It is also necessary to know if it was created or not so the appropriate feeds can be updated